File: src/main/java/com/hmo/cyclism/config/DataImportConfiguration.java
package com.hmo.cyclism.config;

import com.hmo.cyclism.model.Cyclist;
import org.springframework.boot.CommandLineRunner;
import org.springframework.context.annotation.Bean;
import org.springframework.data.mongodb.core.MongoOperations;
import org.springframework.stereotype.Component;

import java.util.Arrays;
import java.util.List;

@Component
public class DataImportConfiguration {

    @Bean
    public CommandLineRunner initData(MongoOperations mongo) {
        return (String... args) -> {
            mongo.dropCollection(Cyclist.class);
            mongo.createCollection(Cyclist.class);
            getCyclists().forEach(mongo::save);
        };
    }

    private List<Cyclist> getCyclists() {
        return Arrays.asList(
                new Cyclist("Chris Froome", "Ineos"),
                new Cyclist("Vincenzo Nibali", "Bahrain Merida"),
                new Cyclist("Thibaut Pinot", "Groupama FDJ"),
                new Cyclist("Richie Porte", "Trek Segafredo"),
                new Cyclist("Primoz Roglic", "Jumbo Visma"));
    }
}

File: src/main/kotlin/com/hmo/cyclism/config/DataImportConfiguration.kt
package com.hmo.cyclism.config

import com.hmo.cyclism.model.Cyclist
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.CommandLineRunner
import org.springframework.data.mongodb.core.ReactiveMongoOperations
import org.springframework.stereotype.Component


@Component
class DataImportConfiguration(@Autowired val mongo: ReactiveMongoOperations) : CommandLineRunner {
    override fun run(vararg args: String?) {
 
        mongo.collectionExists(Cyclist::class.java)
                .doOnNext { col ->
                    if (!col)
                        mongo.createCollection(Cyclist::class.java).subscribe()
                    getCyclists().forEach { c -> mongo.save(c).subscribe() }


                }.subscribe()


    }

    fun getCyclists(): List<Cyclist> {
        return listOf(
                Cyclist("", "Chris Froome", "Ineos"),
                Cyclist("", "Vincenzo Nibali", "Bahrain Merida"),
                Cyclist("", "Thibaut Pinot", "Groupama FDJ"),
                Cyclist("", "Richie Porte", "Trek Segafredo"),
                Cyclist("", "Primoz Roglic", "Jumbo Visma")
        )
    }

}